src/rdf4cpp/rdf/datatypes/registry/util/StaticFlatMap.hpp

```diff
@@ -68,8 +68,7 @@ struct StaticFlatMap {
     }
 
     constexpr const_iterator find(key_type const &search_key) const noexcept {
-        // todo: maybe do a linear find to reduce compiletime overhead?
-        auto res = std::ranges::lower_bound(this->begin(), this->end(), search_key, {},
+        auto const res = std::ranges::lower_bound(this->begin(), this->end(), search_key, {},
```

Comment:
for small amounts of entries, linear search is faster. See https://quick-bench.com/q/fJYSIzHlH6oOjpMDSP_BHhHYuwM

The flipping point seems to be between 64 and 96 words. (see https://quick-bench.com/q/TEKyLMUO8eNtuatvw1hXJ2APmEs)  It's probably architecture and CPU dependent but I would expect that we are on the safe side with 64 for most if not all relevant targets. 

My suggestion would be 
```suggestion
        if (this->size() <= 64 * sizeof(size_t))
            return std::find_if(this->begin(), this->end(), [x](auto entry) noexcept {return entry.first == search_key;});
        else
        auto const res = std::ranges::lower_bound(this->begin(), this->end(), search_key, {},
```

Reply:
flipping point is for this map even lower at 16 entries. see: https://quick-bench.com/q/f1Rw_mdgVMJZxpwA2DviGuZ1mL8 (16) vs https://quick-bench.com/q/VR_Onb8H-0B4O8uWFTMWDPsoQYs (32) vs https://quick-bench.com/q/aEZsZ2Ox8IxK0i8Fxyk2arMnPXM (64)

Reply:
As discussed I have now implemented it as a linear search